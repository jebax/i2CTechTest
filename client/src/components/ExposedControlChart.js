import React from 'react'
import PropTypes from 'prop-types'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import {
  XYPlot,
  XAxis,
  YAxis,
  VerticalGridLines,
  HorizontalGridLines,
  LineMarkSeries,
  DiscreteColorLegend
} from 'react-vis'

const ExposedControlChart = ({ exposed, control, title }) => {
  return (
    <Paper
      elevation={3}
      data-id='chartCard'
    >
      <div
        className='chartWrapper'
      >
        <Typography
          variant='h6'
          color='inherit'
        >
          Weekly Data: {title}
        </Typography>
        <XYPlot
          height={300}
          width={350}
          xType={'time'}
          margin={{
            left: 60
          }}
        >
          <HorizontalGridLines />
          <VerticalGridLines />
          <XAxis
            tickLabelAngle={-30}/>
          <YAxis
            tickLabelAngle={-30}
          />
          <LineMarkSeries
            className={'exposedData'}
            curve={'curveMonotoneX'}
            data={exposed}
          />
          <LineMarkSeries
            className={'controlData'}
            curve={'curveMonotoneX'}
            data={control}
          />
        </XYPlot>
        <DiscreteColorLegend
          orientation='horizontal'
          items={['Exposed', 'Control']}
        />
      </div>
    </Paper>
  )
}

ExposedControlChart.propTypes = {
  exposed: PropTypes.array.isRequired,
  control: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired
}

export default ExposedControlChart
