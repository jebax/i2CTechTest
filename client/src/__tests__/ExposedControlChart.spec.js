import React from 'react'
import { shallow } from 'enzyme'
import ExposedControlChart from '../components/ExposedControlChart'

describe('exposedControl chart', () => {
  var exposedControlChart
  var exposed
  var control

  beforeAll(() => {
    exposed = [{ x: new Date('2018-01-01'), y: '444' }]
    control = [{ x: new Date('2018-01-01'), y: '333' }]

    exposedControlChart = shallow(
      <ExposedControlChart
        exposed={exposed}
        control={control}
      />
    )
  })

  it('renders correctly', () => {
    expect(exposedControlChart).toMatchSnapshot()
  })

  it('renders a react-vis chart', () => {
    expect(exposedControlChart.find('XYPlot').length).toEqual(1)
  })

  it('displays time on the x-axis', () => {
    const plot = exposedControlChart.find('XYPlot')
    expect(plot.props().xType).toEqual('time')
  })

  it('renders two LineMarkSeries components', () => {
    expect(exposedControlChart.find('LineMarkSeries').length).toEqual(2)
  })

  it('renders the exposed data correctly', () => {
    const firstLine = exposedControlChart.find('LineMarkSeries').first()

    expect(firstLine.props().className).toEqual('exposedData')
    expect(firstLine.props().data).toEqual(exposed)
  })

  it('renders the control data correctly', () => {
    const secondLine = exposedControlChart.find('LineMarkSeries').at(1)

    expect(secondLine.props().className).toEqual('controlData')
    expect(secondLine.props().data).toEqual(control)
  })
})
