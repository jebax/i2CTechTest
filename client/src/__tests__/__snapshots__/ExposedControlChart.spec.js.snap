// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exposedControl chart renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExposedControlChart
    control={
      Array [
        Object {
          "x": 2018-01-01T00:00:00.000Z,
          "y": "333",
        },
      ]
    }
    exposed={
      Array [
        Object {
          "x": 2018-01-01T00:00:00.000Z,
          "y": "444",
        },
      ]
    }
    title="TestTitle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="chartWrapper"
      >
        <WithStyles(Typography)
          color="inherit"
          variant="h6"
        >
          Weekly Data: 
          TestTitle
        </WithStyles(Typography)>
        <XYPlot
          className=""
          height={300}
          margin={
            Object {
              "left": 60,
            }
          }
          width={350}
          xType="time"
        >
          <HorizontalGridLines
            attr="y"
            direction="horizontal"
          />
          <VerticalGridLines
            attr="x"
            direction="vertical"
          />
          <XAxis
            attr="x"
            attrAxis="y"
            orientation="bottom"
            tickLabelAngle={-30}
          />
          <YAxis
            attr="y"
            attrAxis="x"
            orientation="left"
            tickLabelAngle={-30}
          />
          <LineMarkSeries
            className="exposedData"
            curve="curveMonotoneX"
            data={
              Array [
                Object {
                  "x": 2018-01-01T00:00:00.000Z,
                  "y": "444",
                },
              ]
            }
            getNull={[Function]}
            lineStyle={Object {}}
            markStyle={Object {}}
            opacity={1}
            stack={false}
            strokeStyle="solid"
            style={Object {}}
          />
          <LineMarkSeries
            className="controlData"
            curve="curveMonotoneX"
            data={
              Array [
                Object {
                  "x": 2018-01-01T00:00:00.000Z,
                  "y": "333",
                },
              ]
            }
            getNull={[Function]}
            lineStyle={Object {}}
            markStyle={Object {}}
            opacity={1}
            stack={false}
            strokeStyle="solid"
            style={Object {}}
          />
        </XYPlot>
        <DiscreteColorLegendItem
          className=""
          colors={
            Array [
              "#12939A",
              "#79C7E3",
              "#1A3177",
              "#FF9833",
              "#EF5D28",
            ]
          }
          items={
            Array [
              "Exposed",
              "Control",
            ]
          }
          orientation="vertical"
        />
      </div>,
      "data-id": "exposedControlChart",
      "elevation": 3,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            color="inherit"
            variant="h6"
          >
            Weekly Data: 
            TestTitle
          </WithStyles(Typography)>,
          <XYPlot
            className=""
            height={300}
            margin={
              Object {
                "left": 60,
              }
            }
            width={350}
            xType="time"
          >
            <HorizontalGridLines
              attr="y"
              direction="horizontal"
            />
            <VerticalGridLines
              attr="x"
              direction="vertical"
            />
            <XAxis
              attr="x"
              attrAxis="y"
              orientation="bottom"
              tickLabelAngle={-30}
            />
            <YAxis
              attr="y"
              attrAxis="x"
              orientation="left"
              tickLabelAngle={-30}
            />
            <LineMarkSeries
              className="exposedData"
              curve="curveMonotoneX"
              data={
                Array [
                  Object {
                    "x": 2018-01-01T00:00:00.000Z,
                    "y": "444",
                  },
                ]
              }
              getNull={[Function]}
              lineStyle={Object {}}
              markStyle={Object {}}
              opacity={1}
              stack={false}
              strokeStyle="solid"
              style={Object {}}
            />
            <LineMarkSeries
              className="controlData"
              curve="curveMonotoneX"
              data={
                Array [
                  Object {
                    "x": 2018-01-01T00:00:00.000Z,
                    "y": "333",
                  },
                ]
              }
              getNull={[Function]}
              lineStyle={Object {}}
              markStyle={Object {}}
              opacity={1}
              stack={false}
              strokeStyle="solid"
              style={Object {}}
            />
          </XYPlot>,
          <DiscreteColorLegendItem
            className=""
            colors={
              Array [
                "#12939A",
                "#79C7E3",
                "#1A3177",
                "#FF9833",
                "#EF5D28",
              ]
            }
            items={
              Array [
                "Exposed",
                "Control",
              ]
            }
            orientation="vertical"
          />,
        ],
        "className": "chartWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Weekly Data: ",
              "TestTitle",
            ],
            "color": "inherit",
            "variant": "h6",
          },
          "ref": null,
          "rendered": Array [
            "Weekly Data: ",
            "TestTitle",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <HorizontalGridLines
                attr="y"
                direction="horizontal"
              />,
              <VerticalGridLines
                attr="x"
                direction="vertical"
              />,
              <XAxis
                attr="x"
                attrAxis="y"
                orientation="bottom"
                tickLabelAngle={-30}
              />,
              <YAxis
                attr="y"
                attrAxis="x"
                orientation="left"
                tickLabelAngle={-30}
              />,
              <LineMarkSeries
                className="exposedData"
                curve="curveMonotoneX"
                data={
                  Array [
                    Object {
                      "x": 2018-01-01T00:00:00.000Z,
                      "y": "444",
                    },
                  ]
                }
                getNull={[Function]}
                lineStyle={Object {}}
                markStyle={Object {}}
                opacity={1}
                stack={false}
                strokeStyle="solid"
                style={Object {}}
              />,
              <LineMarkSeries
                className="controlData"
                curve="curveMonotoneX"
                data={
                  Array [
                    Object {
                      "x": 2018-01-01T00:00:00.000Z,
                      "y": "333",
                    },
                  ]
                }
                getNull={[Function]}
                lineStyle={Object {}}
                markStyle={Object {}}
                opacity={1}
                stack={false}
                strokeStyle="solid"
                style={Object {}}
              />,
            ],
            "className": "",
            "height": 300,
            "margin": Object {
              "left": 60,
            },
            "width": 350,
            "xType": "time",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "attr": "y",
                "direction": "horizontal",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "attr": "x",
                "direction": "vertical",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "attr": "x",
                "attrAxis": "y",
                "orientation": "bottom",
                "tickLabelAngle": -30,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "attr": "y",
                "attrAxis": "x",
                "orientation": "left",
                "tickLabelAngle": -30,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "exposedData",
                "curve": "curveMonotoneX",
                "data": Array [
                  Object {
                    "x": 2018-01-01T00:00:00.000Z,
                    "y": "444",
                  },
                ],
                "getNull": [Function],
                "lineStyle": Object {},
                "markStyle": Object {},
                "opacity": 1,
                "stack": false,
                "strokeStyle": "solid",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "controlData",
                "curve": "curveMonotoneX",
                "data": Array [
                  Object {
                    "x": 2018-01-01T00:00:00.000Z,
                    "y": "333",
                  },
                ],
                "getNull": [Function],
                "lineStyle": Object {},
                "markStyle": Object {},
                "opacity": 1,
                "stack": false,
                "strokeStyle": "solid",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "colors": Array [
              "#12939A",
              "#79C7E3",
              "#1A3177",
              "#FF9833",
              "#EF5D28",
            ],
            "items": Array [
              "Exposed",
              "Control",
            ],
            "orientation": "vertical",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="chartWrapper"
        >
          <WithStyles(Typography)
            color="inherit"
            variant="h6"
          >
            Weekly Data: 
            TestTitle
          </WithStyles(Typography)>
          <XYPlot
            className=""
            height={300}
            margin={
              Object {
                "left": 60,
              }
            }
            width={350}
            xType="time"
          >
            <HorizontalGridLines
              attr="y"
              direction="horizontal"
            />
            <VerticalGridLines
              attr="x"
              direction="vertical"
            />
            <XAxis
              attr="x"
              attrAxis="y"
              orientation="bottom"
              tickLabelAngle={-30}
            />
            <YAxis
              attr="y"
              attrAxis="x"
              orientation="left"
              tickLabelAngle={-30}
            />
            <LineMarkSeries
              className="exposedData"
              curve="curveMonotoneX"
              data={
                Array [
                  Object {
                    "x": 2018-01-01T00:00:00.000Z,
                    "y": "444",
                  },
                ]
              }
              getNull={[Function]}
              lineStyle={Object {}}
              markStyle={Object {}}
              opacity={1}
              stack={false}
              strokeStyle="solid"
              style={Object {}}
            />
            <LineMarkSeries
              className="controlData"
              curve="curveMonotoneX"
              data={
                Array [
                  Object {
                    "x": 2018-01-01T00:00:00.000Z,
                    "y": "333",
                  },
                ]
              }
              getNull={[Function]}
              lineStyle={Object {}}
              markStyle={Object {}}
              opacity={1}
              stack={false}
              strokeStyle="solid"
              style={Object {}}
            />
          </XYPlot>
          <DiscreteColorLegendItem
            className=""
            colors={
              Array [
                "#12939A",
                "#79C7E3",
                "#1A3177",
                "#FF9833",
                "#EF5D28",
              ]
            }
            items={
              Array [
                "Exposed",
                "Control",
              ]
            }
            orientation="vertical"
          />
        </div>,
        "data-id": "exposedControlChart",
        "elevation": 3,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              color="inherit"
              variant="h6"
            >
              Weekly Data: 
              TestTitle
            </WithStyles(Typography)>,
            <XYPlot
              className=""
              height={300}
              margin={
                Object {
                  "left": 60,
                }
              }
              width={350}
              xType="time"
            >
              <HorizontalGridLines
                attr="y"
                direction="horizontal"
              />
              <VerticalGridLines
                attr="x"
                direction="vertical"
              />
              <XAxis
                attr="x"
                attrAxis="y"
                orientation="bottom"
                tickLabelAngle={-30}
              />
              <YAxis
                attr="y"
                attrAxis="x"
                orientation="left"
                tickLabelAngle={-30}
              />
              <LineMarkSeries
                className="exposedData"
                curve="curveMonotoneX"
                data={
                  Array [
                    Object {
                      "x": 2018-01-01T00:00:00.000Z,
                      "y": "444",
                    },
                  ]
                }
                getNull={[Function]}
                lineStyle={Object {}}
                markStyle={Object {}}
                opacity={1}
                stack={false}
                strokeStyle="solid"
                style={Object {}}
              />
              <LineMarkSeries
                className="controlData"
                curve="curveMonotoneX"
                data={
                  Array [
                    Object {
                      "x": 2018-01-01T00:00:00.000Z,
                      "y": "333",
                    },
                  ]
                }
                getNull={[Function]}
                lineStyle={Object {}}
                markStyle={Object {}}
                opacity={1}
                stack={false}
                strokeStyle="solid"
                style={Object {}}
              />
            </XYPlot>,
            <DiscreteColorLegendItem
              className=""
              colors={
                Array [
                  "#12939A",
                  "#79C7E3",
                  "#1A3177",
                  "#FF9833",
                  "#EF5D28",
                ]
              }
              items={
                Array [
                  "Exposed",
                  "Control",
                ]
              }
              orientation="vertical"
            />,
          ],
          "className": "chartWrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Weekly Data: ",
                "TestTitle",
              ],
              "color": "inherit",
              "variant": "h6",
            },
            "ref": null,
            "rendered": Array [
              "Weekly Data: ",
              "TestTitle",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <HorizontalGridLines
                  attr="y"
                  direction="horizontal"
                />,
                <VerticalGridLines
                  attr="x"
                  direction="vertical"
                />,
                <XAxis
                  attr="x"
                  attrAxis="y"
                  orientation="bottom"
                  tickLabelAngle={-30}
                />,
                <YAxis
                  attr="y"
                  attrAxis="x"
                  orientation="left"
                  tickLabelAngle={-30}
                />,
                <LineMarkSeries
                  className="exposedData"
                  curve="curveMonotoneX"
                  data={
                    Array [
                      Object {
                        "x": 2018-01-01T00:00:00.000Z,
                        "y": "444",
                      },
                    ]
                  }
                  getNull={[Function]}
                  lineStyle={Object {}}
                  markStyle={Object {}}
                  opacity={1}
                  stack={false}
                  strokeStyle="solid"
                  style={Object {}}
                />,
                <LineMarkSeries
                  className="controlData"
                  curve="curveMonotoneX"
                  data={
                    Array [
                      Object {
                        "x": 2018-01-01T00:00:00.000Z,
                        "y": "333",
                      },
                    ]
                  }
                  getNull={[Function]}
                  lineStyle={Object {}}
                  markStyle={Object {}}
                  opacity={1}
                  stack={false}
                  strokeStyle="solid"
                  style={Object {}}
                />,
              ],
              "className": "",
              "height": 300,
              "margin": Object {
                "left": 60,
              },
              "width": 350,
              "xType": "time",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "attr": "y",
                  "direction": "horizontal",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "attr": "x",
                  "direction": "vertical",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "attr": "x",
                  "attrAxis": "y",
                  "orientation": "bottom",
                  "tickLabelAngle": -30,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "attr": "y",
                  "attrAxis": "x",
                  "orientation": "left",
                  "tickLabelAngle": -30,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "exposedData",
                  "curve": "curveMonotoneX",
                  "data": Array [
                    Object {
                      "x": 2018-01-01T00:00:00.000Z,
                      "y": "444",
                    },
                  ],
                  "getNull": [Function],
                  "lineStyle": Object {},
                  "markStyle": Object {},
                  "opacity": 1,
                  "stack": false,
                  "strokeStyle": "solid",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "controlData",
                  "curve": "curveMonotoneX",
                  "data": Array [
                    Object {
                      "x": 2018-01-01T00:00:00.000Z,
                      "y": "333",
                    },
                  ],
                  "getNull": [Function],
                  "lineStyle": Object {},
                  "markStyle": Object {},
                  "opacity": 1,
                  "stack": false,
                  "strokeStyle": "solid",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "colors": Array [
                "#12939A",
                "#79C7E3",
                "#1A3177",
                "#FF9833",
                "#EF5D28",
              ],
              "items": Array [
                "Exposed",
                "Control",
              ],
              "orientation": "vertical",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
